portfolio/src/main/java/com/google/sps/servlets/ListCommentServlet.java

```diff
@@ -0,0 +1,115 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import com.google.sps.utils.CommentDataStore;
+import com.google.sps.utils.CommentUtils;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/** Servlet that handles posting comment content. */
+@WebServlet("/list-comment")
+public class ListCommentServlet extends HttpServlet {
+    private int maxNumberOfComments;
+
+    static class Constants {
+        final static String MAXNUMBER = "quantity";
+    }
+
+    @Override
+    public void init() {
+        this.maxNumberOfComments = 0; // default value
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Gets the displayed comment limit
+        getMaxNumberOfComments(request, response);
+
+        // Loads the comment from Datastore
+        List<Comment> comments = getComments(request);
+
+        // Converts into json form
+        String json = convertToJsonUsingGson(comments);
+
+        // Sends the JSON as the response
+        sendJsonResponse(response, json);
+    }
+
+    private void getMaxNumberOfComments(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
The name of this method is get but implementation is more like a set as it sets state and return void?
Maybe just let it return the value without setting anything?

Reply:
Fixed